test/Npgsql.Tests/Types/TsQueryTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections;
+using System.Threading.Tasks;
+using NpgsqlTypes;
+using NUnit.Framework;
+
+namespace Npgsql.Tests.Types
+{
+    [TestFixture(MultiplexingMode.NonMultiplexing, false)]
+    [TestFixture(MultiplexingMode.NonMultiplexing, true)]
+    [TestFixture(MultiplexingMode.Multiplexing, false)]
+    [TestFixture(MultiplexingMode.Multiplexing, true)]
+    public sealed class TsQueryTests : TypeHandlerTestBase<NpgsqlTsQuery>
+    {
+        public TsQueryTests(MultiplexingMode multiplexingMode, bool useTypeName) : base(
+            multiplexingMode,
+            useTypeName ? null : NpgsqlDbType.TsQuery,
+            useTypeName ? "tsquery" : null)
+        { }
+
+        public static IEnumerable TestCases() => new[]
+        {
+            new object[] {
+                "$$'a' <-> 'b'$$::tsquery",
+                new NpgsqlTsQueryFollowedBy(
+                    new NpgsqlTsQueryLexeme("a"), 1, new NpgsqlTsQueryLexeme("b")) },
+            new object[]{
```

Comment:
This mountain is nice :) But I think we should (in addition) just have one test per TsQuery node type, to quickly understand where's the problem.